packages/x-charts/src/internals/defaultizeColor.ts

```diff
@@ -1,9 +1,18 @@
 import { AllSeriesType } from '../models/seriesType';
 
-export function defaultizeColor(series: AllSeriesType, seriesIndex: number, colors: string[]) {
-  if (series.type === 'pie') {
+export function defaultizeColor<T extends AllSeriesType>(
+  series: T,
+  seriesIndex: number,
+  colors: string[],
+): T {
+  if (
+    series.type === 'pie' ||
+    // @ts-ignore funnel is a pro feature
```

Comment:
IMO we should generally prefer `@ts-expect-error` so that we will be alerted once we fix these type issues

Reply:
We can't because this issue happens on `pro` only. On `community` there is no issue, so `@ts-expect-error` gives an error that is unnecessary.